R-packages/evalcast/tests/testthat/test-evaluate.R

```diff
@@ -0,0 +1,216 @@
+    pcard <- create_pcard(tibble(
+      ahead = 1,
+      geo_value = rep(c("al", "wy"), each=3),
+      quantile = c(0.1, 0.5, 0.9, 0.1, 0.5, 0.9),
+      value = seq(1, 6),
+      forecaster = rep(c("a", "b"), each=3),
+      forecast_date = rep(as.Date(c("2020-01-02", "2020-01-09")), each=3),
+      data_source = "source",
+      signal = "signal",
+      target_end_date = as.Date("2020-01-21"),
+      incidence_period = "epiweek"))
+
+    side_truth <- tibble(
+      forecaster = c("a", "b"),
+      actual = c(5, 10)
+    )
+    score_card <- evaluate_predictions(pcard, side_truth=side_truth, grp_vars=c("forecaster"))
+
+    expect_called(mock_download_signal, 0)
+
+    expect_equal(colnames(score_card),
+                  c("wis", "ae", "coverage_80", "forecaster",  "ahead", "geo_value",
+                    "quantile", "value", "forecast_date", "data_source", "signal",
+                    "target_end_date", "incidence_period", "actual"))
+    n <- 6
+    expect_equal(score_card$ahead, rep(1, n))
+    expect_equal(score_card$geo_value, rep(c("al", "wy"), each=n/2))
+    expect_equal(score_card$quantile, rep(c(0.1, 0.5, 0.9), n/3))
+    expect_equal(score_card$value, seq(1, 6))
+    expect_equal(score_card$forecaster, rep(c("a", "b"), each=n/2))
+    expect_equal(score_card$forecast_date, rep(as.Date(c("2020-01-02", "2020-01-09")), each=n/2))
+    expect_equal(score_card$data_source, rep("source", n))
+    expect_equal(score_card$signal, rep("signal", n))
+    expect_equal(score_card$target_end_date, rep(as.Date("2020-01-21"), n))
+    expect_equal(score_card$incidence_period, rep("epiweek", n))
+    expect_equal(score_card$actual, rep(c(5, 10), each=n/2))
+    expect_equal(score_card$wis, rep(c(7.4/3, 13.4/3), each=n/2)) # computed by hand
+    expect_equal(score_card$ae, rep(c(3, 5), each=n/2))
+    expect_equal(score_card$coverage_80, rep(0, n))
+  })  
+})
+
+test_that("evaluate_predictions backfill_buffer works", {
+  # Mock `download_signals()` so we don't call to the covidcast API.
+  mock_download_signal <- mock(create_fake_downloaded_signal(c("al", "wy"), 10), cycle=TRUE)
+  mockr::with_mock(download_signal = mock_download_signal, {
+    pcard <- create_pcard(tibble(
+      ahead = 1,
+      geo_value = rep(c("al", "wy"), each=3),
+      quantile = c(0.1, 0.5, 0.9, 0.1, 0.5, 0.9),
+      value = seq(1, 6),
+      forecaster = "a",
+      # The backfill buffer compares against today's date, so we need to set the forecast date
+      # dynamically here or else after n days these tests will stop working.
+      forecast_date = Sys.Date() - 21,
+      data_source = "source",
+      signal = "signal",
+      target_end_date = as.Date("2020-01-05"),
+      incidence_period = "epiweek"
+    ))
+    expect_warning(evaluate_predictions(pcard, backfill_buffer = 22),
+                   "backfill_buffer")
+    # waited long enough should have no warning:
+    evaluate_predictions(pcard, backfill_buffer = 7)
+  })
```

Comment:
I don't really understand how any of this works, but I assume it is actually testing what we want!

Reply:
This is copied (more or less) from the existing skipped tests.  AFAICT we are testing whether the `backfill_buffer` argument does anything, but right now the only thing it does is raise a warning.  So we shouldn't get any warnings when all the dates are in the buffer.  I have explicitly added a check that creates an error when there are warnings on this run.  Ultimately we don't care what the output is here, but rather just that moving the buffer forwards or backwards has some impact.